rise/static/main.js

```diff
@@ -255,68 +277,61 @@ define([
    * changing to the one we want. By changing the URL before setting up reveal,
    * the slideshow really starts on the desired slide.
    */
-  function setStartingSlide(selected, config) {
-
-    var start_slideshow_promise = config.get('start_slideshow_at');
-    // We need the value after the promise resolution
-    start_slideshow_promise.then(function(start_slideshow){
-      if (start_slideshow === 'selected') {
-        // Start from the selected cell
-        Reveal.slide(selected[0], selected[1]);
-      } else {
-        // Start from the beginning
-        Reveal.slide(0, 0);
-      }
-      setScrollingSlide(config);
-    });
+  function setStartingSlide(selected) {
 
+    var start_slideshow = complete_config.start_slideshow_at;
+    if (start_slideshow === 'selected') {
+      // Start from the selected cell
+      Reveal.slide(selected[0], selected[1]);
+    } else {
+      // Start from the beginning
+      Reveal.slide(0, 0);
+    }
+    setScrollingSlide();
   }
 
   /* Setup the scrolling in the current slide if the config option is activated
    *  and the content is greater than 0.95 * slide height
    */
-  function setScrollingSlide(config) {
-
-    var scroll_promise = config.get('scroll');
-    scroll_promise.then(function(scroll){
-      if (scroll === true) {
-        var h = $('.reveal').height() * 0.95;
-        $('section.present').find('section')
-          .filter(function() {
-            return $(this).height() > h;
-          })
-          .css('height', 'calc(95vh)')
-          .css('overflow-y', 'scroll')
-          .css('margin-top', '20px');
-      }
-    });
-
+  function setScrollingSlide() {
+
+    var scroll = complete_config.scroll;
+    if (scroll === true) {
+      var h = $('.reveal').height() * 0.95;
+      $('section.present').find('section')
+        .filter(function() {
+          return $(this).height() > h;
+        })
+        .css('height', 'calc(95vh)')
+        .css('overflow-y', 'scroll')
+        .css('margin-top', '20px');
+    }
   }
 
-  /* Setup the auto-launch function, which checks metadata to see if
+  /* 
+   * Setup the auto-launch function, which checks metadata to see if
    * RISE should launch automatically when the notebook is opened.
+   * 
+   * this will trigger only on notebooks that have 
+   * either a 'livereveal' or a 'rise' section in their metadata
+   * this is because autolaunch can be enabled in nbextensions_configurator
+   * and so can possibly have a too big impact if we are not careful
    */
-  function autoLaunch(config) {
-
-    var autolaunch_promise = config.get('autolaunch');
-    autolaunch_promise.then(function(autolaunch){
-      if (autolaunch === true) {
-        revealMode();
-      }
-    });
-
+  function autoLaunch() {
+    if (complete_config.autolaunch && complete_config.is_slideshow)
```

Comment:
Why are you waiting here for `complete_config.is_slideshow`?

Reply:
But `complete_config.is_slideshow` exists only if the notebook has some rise metadata defined. What happens with the slideshows that do not have any rise metadata defined?

Reply:
Well unfortunately these notebooks won't autolaunch. I thought that would be an acceptable tradeoff.

Now that we talk about it: another approach would be to simply look for cells that are tagged as 'Slide'; that would be the criteria behind 'is_slideshow', and it would allow to more effectively classify notebooks that need to autolaunch

How does that sound ?

Reply:
> How does that sound ?

Sounds good to me! I was thinking the same when I reviewed the PR but I forgot to write it down.